// @gromo-fintech/log4g/dist/transport-config/TransportConfiguratorBasic.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransportConfiguratorBasic = void 0;
const winston_1 = require("winston");
const winston_daily_rotate_file_1 = __importDefault(require("winston-daily-rotate-file"));
const FormatterBasic_1 = require("../formatter/FormatterBasic");
const LogLevel_enum_1 = require("../logger/enums/LogLevel.enum");
const FormatterAdvance_1 = require("../formatter/FormatterAdvance");
class TransportConfiguratorBasic {
    configureTransports(options) {
        const transportList = [];
        const formatLog = new FormatterBasic_1.FormatLogBasic();
        const advanceFormatLog = new FormatterAdvance_1.FormatLogWithBifurcation();
        const loggerFormat = formatLog.formatter(options.logFormat, LogLevel_enum_1.LogLevel.HTTP);
        if (options.enableStdout) {
            transportList.push(new winston_1.transports.Console({
                format: loggerFormat,
            }));
        }
        if (options.fileOptions && options.fileOptions.enableFile) {
            if (!options.nameOfProject || options.nameOfProject.trim() === '') {
                throw new Error("File logging is enabled but no project name was provided.");
            }
            const fileName = `${process.env.HOME}/.gromo-logger/APPLICATION_LOG/${options.nameOfProject}-logs`;
            transportList.push(new winston_daily_rotate_file_1.default({
                filename: fileName,
                datePattern: options.fileOptions.datePattern,
                zippedArchive: options.fileOptions.zippedArchive,
                maxSize: options.fileOptions.maxSize,
                maxFiles: options.fileOptions.maxDuration,
                format: loggerFormat,
            }));
        }
        if (options.enableAccessLog) {
            const advaceFormatter = advanceFormatLog.formatter(options.logFormat, LogLevel_enum_1.LogLevel.HTTP);
            if (!options.nameOfProject || options.nameOfProject.trim() === '') {
                throw new Error("File logging is enabled but no project name was provided.");
            }
            const fileName = `${process.env.HOME}/.gromo-logger/ACCESS_LOG/${options.nameOfProject}-logs`;
            transportList.push(new winston_daily_rotate_file_1.default({
                filename: fileName,
                datePattern: options.fileOptions.datePattern,
                zippedArchive: options.fileOptions.zippedArchive,
                maxSize: options.fileOptions.maxSize,
                maxFiles: options.fileOptions.maxDuration,
                format: advaceFormatter,
            }));
        }
        return transportList;
    }
}
exports.TransportConfiguratorBasic = TransportConfiguratorBasic;
