// @gromo-fintech/log4g/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FastifyMiddleware = exports.ExpressMiddleware = exports.logger = exports.TransporterType = exports.LogLevel = exports.LogFormat = void 0;
var logFormat_enum_1 = require("./formatter/enums/logFormat.enum");
Object.defineProperty(exports, "LogFormat", { enumerable: true, get: function () { return logFormat_enum_1.LogFormat; } });
var LogLevel_enum_1 = require("./logger/enums/LogLevel.enum");
Object.defineProperty(exports, "LogLevel", { enumerable: true, get: function () { return LogLevel_enum_1.LogLevel; } });
var TransporterType_enum_1 = require("./transport-config/enums/TransporterType.enum");
Object.defineProperty(exports, "TransporterType", { enumerable: true, get: function () { return TransporterType_enum_1.TransporterType; } });
var GroMoLogger_1 = require("./logger/GroMoLogger");
Object.defineProperty(exports, "logger", { enumerable: true, get: function () { return __importDefault(GroMoLogger_1).default; } });
__exportStar(require("./interceptors/nestInterceptor"), exports);
var ExpressMiddleware_1 = require("./interceptors/ExpressMiddleware");
Object.defineProperty(exports, "ExpressMiddleware", { enumerable: true, get: function () { return ExpressMiddleware_1.ExpressMiddleware; } });
var FastifyMiddleware_1 = require("./interceptors/FastifyMiddleware");
Object.defineProperty(exports, "FastifyMiddleware", { enumerable: true, get: function () { return FastifyMiddleware_1.FastifyMiddleware; } });
